### PR TITLE
feat: Add support for tokens in the XRootD interface

### PR DESCRIPTION
**Pull request by bcouturi**
### Description

Support for tokens in the interface with XRootD.
As the tokens used by EOS at CERN are URL parameters, this PR allows adding them when needed (an extra "decorator" parameter has been introduced) 

### QC

* [ ] The PR contains a test case for the changes or the changes are already covered by an existing test case.
* [x ] The documentation (`docs/`) is updated to reflect the changes or this is not necessary (e.g. if the change does neither modify the language nor the behavior or functionalities of Snakemake).

### CODE REVIEW

File: docs/snakefiles/remote_files.rst [hunk 1]
```diff
@@ -510,6 +510,14 @@ This flag can be overridden on a file by file basis as described in the S3 remot
         shell:
             'xrdcp {input[0]} {output[0]}'
 
+In order to access the files using autorization tokens, the "decorator" parameter can be used to append the necessary string to the URL e.g.
+
+.. code-block:: python
+
+    from snakemake.remote.XRootD import RemoteProvider as XRootDRemoteProvider
+    XRootD = XRootDRemoteProvider(stay_on_remote=True, decorator=lambda x: x + "?xrd.wantprot=unix&authz=XXXXXX")
```
Comment (johanneskoester) [inline docs/snakefiles/remote_files.rst:518]: I'd probably name this a bit more specific, e.g. `url_decorator`?

**Review by johanneskoester — CHANGES_REQUESTED** [review]
THanks a lot. I small suggestion below.

File: docs/snakefiles/remote_files.rst (hunk 1 above)
Comment (bcouturi) [reply docs/snakefiles/remote_files.rst]: Indeed, that is clearer. I have updated the code and re-formatted.

**Review by johanneskoester — APPROVED** [review]

**Comment by bcouturi** [comment]
The previous tests failed because the change to IO.py was not protected enough in case the NamedList contained str and not _IOFiles. I added a check.

**Comment by johanneskoester** [comment]
May I ask you to format with black?

**Comment by bcouturi** [comment]
Sorry, I thought I had done it already. I have rebased on the snakemake main branch and ran black.

**Comment by bcouturi** [comment]
The CI formatting check had failed in files which are not in my MR, it looks like the latest black changed the default format. 
As those changes are already included in the main branch, I have just rebased.

**Pull request merged**